ocdata/core/bulk.py

```diff
@@ -0,0 +1,287 @@
+            (
+                self.atoms,
+                self.mpid,
+                _,
+                _,
+            ) = bulk_db[bulk_id_from_db]
+        else:
+            bulk_db = pickle.load(open(bulk_db_path, "rb"))
+            bulk_id_from_db = np.random.randint(len(bulk_db))
+            (
+                self.atoms,
+                self.mpid,
+                _,
+                _,
+            ) = bulk_db[bulk_id_from_db]
+            self.bulk_id_from_db = bulk_id_from_db
+
+        # Comment(@abhshkdz): Do we need this?
+        self.n_elems = len(set(self.atoms.symbols))
+
+    def set_mpid(self, mpid: str):
+        self.mpid = mpid
+
+    def set_bulk_id_from_db(self, bulk_id_from_db: int):
+        self.bulk_id_from_db = bulk_id_from_db
+
+    def get_slabs(self):
+        """
+        Returns a list of possible slabs for this bulk instance.
+        This can be later used to iterate through all slabs,
+        or select one at random, to make a Surface object.
+        """
+        if self.precomputed_slabs_path is not None and self.bulk_id_from_db is not None:
+            slabs = self.get_precomputed_slabs()
+        else:
+            slabs = self.compute_slabs()
+
+        return slabs
+
+    def get_precomputed_slabs(self):
+        """
+        Loads relevant pickle of precomputed slabs, and returns
+        a list of tuples containing: (atoms, miller, shift, top).
+        """
+        assert self.bulk_id_from_db is not None
+
+        bulk_slabs_path = os.path.join(
+            self.precomputed_slabs_path, f"{self.bulk_id_from_db}.pkl"
+        )
+        assert os.path.exists(bulk_slabs_path)
+
+        with open(bulk_slabs_path, "rb") as f:
+            slabs_info = pickle.load(f)
+
+        return slabs_info
+
+    def compute_slabs(self, max_miller=MAX_MILLER):
+        """
+        Enumerates all the symmetrically distinct slabs of a bulk structure.
+        It will not enumerate slabs with Miller indices above the
+        `max_miller` argument. Note that we also look at the bottoms of slabs
+        if they are distinct from the top. If they are distinct, we flip the
+        surface so the bottom is pointing upwards.
+
+        Args:
+            max_miller  An integer indicating the maximum Miller index of the slabs
+                        you are willing to enumerate. Increasing this argument will
+                        increase the number of slabs, but the slabs will
+                        generally become larger.
+        Returns:
+            all_slabs_info  A list of 4-tuples containing:  `pymatgen.Structure`
+                            objects for slabs we have enumerated, the Miller
+                            indices, floats for the shifts, and Booleans for "top".
+        """
+        bulk_struct = standardize_bulk(self.atoms)
+
+        all_slabs_info = []
+        for millers in get_symmetrically_distinct_miller_indices(
+            bulk_struct, MAX_MILLER
```

Comment:
Let's use the argument and not global constant - `MAX_MILLER` --> `max_miller`

Reply:
Updated: https://github.com/Open-Catalyst-Project/Open-Catalyst-Dataset/blob/refactor-v2/ocdata/core/slab.py#L386